app/src/main/java/com/github/se/project/ui/components/LessonEditor.kt

```diff
@@ -0,0 +1,287 @@
+  var minPrice by remember { mutableDoubleStateOf(lesson?.minPrice ?: 5.0) }
+  var maxPrice by remember { mutableDoubleStateOf(lesson?.maxPrice ?: 50.0) }
+  var selectedDate by remember { mutableStateOf("") }
+  var selectedTime by remember { mutableStateOf("") }
+
+  val calendar = Calendar.getInstance()
+  val currentDateTime = Calendar.getInstance()
+  val currentLessonId = remember { mutableStateOf<String?>(null) }
+
+  if (currentLessonId.value != lesson?.id) {
+    currentLessonId.value = lesson?.id
+    if (lesson != null) {
+      selectedLanguages.clear()
+      selectedLanguages.addAll(lesson.languages)
+      selectedTime = lesson.timeSlot.split("T")[1].substring(0, 5)
+      selectedDate = lesson.timeSlot.split("T")[0]
+    }
+  }
+
+  // Context for the Toast messages
+  val context = LocalContext.current
+
+  // Date Picker
+  val datePickerDialog =
+      DatePickerDialog(
+          context,
+          { _, year, month, dayOfMonth ->
+            val selectedCalendar = Calendar.getInstance()
+            selectedCalendar.set(year, month, dayOfMonth)
+            selectedDate = "$dayOfMonth/${month + 1}/$year"
+          },
+          calendar.get(Calendar.YEAR),
+          calendar.get(Calendar.MONTH),
+          calendar.get(Calendar.DAY_OF_MONTH))
+  datePickerDialog.datePicker.minDate = currentDateTime.timeInMillis
+
+  // Time Picker
+  val timePickerDialog =
+      TimePickerDialog(
+          context,
+          { _, hourOfDay, minute ->
+            val formattedHour = hourOfDay.toString().padStart(2, '0')
+            val formattedMinute = minute.toString().padStart(2, '0')
+
+            val selectedCalendar =
+                Calendar.getInstance().apply {
+                  timeInMillis = currentDateTime.timeInMillis
+                  set(Calendar.HOUR_OF_DAY, hourOfDay)
+                  set(Calendar.MINUTE, minute)
+                }
+
+            val isSelectedDateToday =
+                selectedDate ==
+                    "${calendar.get(Calendar.DAY_OF_MONTH)}/${calendar.get(Calendar.MONTH) + 1}/${calendar.get(Calendar.YEAR)}"
+
+            if (isSelectedDateToday && selectedCalendar.before(currentDateTime)) {
+              Toast.makeText(context, "You cannot select a past time", Toast.LENGTH_SHORT).show()
+            } else {
+              selectedTime = "$formattedHour:$formattedMinute"
+            }
+          },
+          calendar.get(Calendar.HOUR_OF_DAY),
+          calendar.get(Calendar.MINUTE),
+          true)
+
+  val onConfirmClick = {
+    val error =
+        validateLessonInput(
+            title,
+            description,
+            selectedSubject,
+            selectedLanguages,
+            selectedDate,
+            selectedTime,
+        )
+    if (error != null) {
+      Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
+    } else {
+      onConfirm(
+          Lesson(
+              lesson!!.id,
```

Comment:
This is risky to use lesson!!.id knowing that the lesson you receive in argment is "lesson?" . You probably rather avoid passing "lesson?" to LessonEditor and check before calling LessonEditor that your Lesson can not be null.
One better solution is to use the LessonViewModel to manipulate the selectedLesson instead of passing a lesson as argment that we would know where it comes from